openslides/agenda/projector.py

```diff
@@ -90,35 +96,30 @@ def get_requirements(self, config_entry):
                 # Yield last speakers
                 yield speaker.user
 
-    def get_agenda_items(self, projector_id):
-        projector = Projector.objects.get(pk=projector_id)
+    def get_agenda_items(self, projector):
         for element in projector.elements.values():
             agenda_item_id = element.get('agenda_item_id')
             if agenda_item_id is not None:
                 yield Item.objects.get(pk=agenda_item_id)
 
     def get_collection_elements_required_for_this(self, collection_element, config_entry):
         output = super().get_collection_elements_required_for_this(collection_element, config_entry)
-        # Full update if agenda_item changes because then we may have new
-        # candidates and therefor need new users.
-        items = self.get_agenda_items(config['projector_currentListOfSpeakers_reference'])
-        for item in items:
-            if collection_element == CollectionElement.from_values(item.get_collection_string(), item.pk):
-                output.extend(self.get_requirements_as_collection_elements(config_entry))
-                break
+        # Full update if agenda_item or referenced projector changes because
+        # then we may have new candidates and therefor need new users.
+        reference_projector = Projector.objects.get(
+            pk=config['projector_currentListOfSpeakers_reference'])
+        is_reference_projector = collection_element == CollectionElement.from_values(
+                reference_projector.get_collection_string(),
+                reference_projector.pk)
+        is_config = collection_element.collection_string == 'core/config' and \
+            collection_element.get_instance()['key'] == 'projector_currentListOfSpeakers_reference'
```

Comment:
I think we should not get the instance from the collection element but use something like this:
`collection_element.information.get('changed_config') == 'projector_currentListOfSpeakers_reference'`

And please do not use `\` but braces.